Observer/ProductDeleteAfter.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Klump\Payment\Observer;
+
+use Klump\Payment\Helpers\SyncHelper;
+use Magento\ConfigurableProduct\Model\Product\Type\Configurable;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+
+class ProductDeleteAfter implements ObserverInterface
+{
+    protected $syncHelper;
+
+    public function __construct(SyncHelper $syncHelper)
+    {
+        $this->syncHelper = $syncHelper;
+    }
+
+    public function execute(Observer $observer): void
+    {
+        $product = $observer->getEvent()->getDataObject();
+
+        $data = [];
+
+        if ($product->getTypeId() == Configurable::TYPE_CODE) {
+            $children = $product->getTypeInstance()->getUsedProducts($product);
+            foreach ($children as $child) {
+                $data[] = [
+                    'name'         => $child->getName(),
+                    'product_id'   => $product->getId(),
+                    'variant_id'   => $child->getId(),
+                    'variant_name' => $child->getName(),
+                    'quantity'     => 0,
+                    'image'        => null,
+                    'is_published' => false,
+                    'price'        => $child->getPrice(),
+                    'sku'          => $child->getSku(),
+                ];
+            }
+        } else {
+            $data[] = [
+                'name'         => $product->getName(),
```

Comment:
The data array here is almost the same as the one on line 28, with variant_id and variant_name being the only distinction. Can we build this array at the top and only bring in those variants if they exist?